BrainPortal/app/controllers/neurohub_application_controller.rb

```diff
@@ -31,6 +31,7 @@ class NeurohubApplicationController < ApplicationController
 
   before_action :switch_to_neurohub_layout
   before_action :prepare_invites
+  before_action :prepare_messages
```

Comment:
Now you're asking the rails framework to call `prepare_messages` twice for every request, because the callback is already registered in ApplicationController

Reply:
I checked and it's not called twice, Rails is intelligent enough not to invoke a method twice if you register it several time. But the point is still valid, it's already registered in ApplicationController.

Reply:
Please remove it